pandas/tseries/tests/test_timeseries.py

```diff
@@ -2454,6 +2454,13 @@ def test_intersection_bug_1708(self):
         result = index_1 & index_2
         self.assertEqual(len(result), 0)
 
+    # GH 10699
+    def test_offset_add_datetimeindex(self):
+        from pandas import DateOffset
+        result = DatetimeIndex(['2000-01-01','2000-01-02']) + DateOffset(years=1)
+        exp = DatetimeIndex(['2001-01-01', '2001-01-02'])
+        self.assertTrue(result.equals(exp))
```

Comment:
you can move all of the above tests (that you have in ``test_series``) here. Then loop thru both a ``Series`` and ``DatetimeIndex`` to do all of the ops. We do a lot of this in ``tseries/tests/test_base.py`` for example.